File: src/protocol/node/state/Policy.ts
/*
 * Filename: Policy.ts
 * FullPath: apps/CWSP-reborn/src/protocol/node/state/Policy.ts
 * Change date and time: 16.30.00_10.07.2026
 * Reason for changes: Thin NodeJS facade re-exporting canonical v2 policy state helpers.
 *
 * NOTE: Re-export only. Endpoint policy state (dedupe, stale, reconnect buffering,
 * clipboard echo suppression) is evaluated by the shared canonical policy module.
 */

export {
    evaluateCwspPolicy,
    evaluatePacketPolicy,
    classifyCwspPacket,
    DEFAULT_CWSP_POLICY,
    DEFAULT_CWSP_RECONNECT_QUEUE_SIZE,
} from "../../../../../../modules/projects/cwsp-shared/src/v2/policy.ts";

export type {
    CwspPolicyConfig,
    CwspPolicyContext,
    CwspPolicyDecision,
    CwspPolicyClass,
    CwspPolicyAction,
    CwspPolicyReason,
} from "../../../../../../modules/projects/cwsp-shared/src/v2/types.ts";

File: src/protocol/java/state/Policy.java
/*
 * Filename: Policy.java
 * FullPath: apps/CWSP-reborn/src/protocol/java/state/Policy.java
 * Change date and time: 16.42.00_10.07.2026
 * Reason for changes: Thin re-export wrapper so callers can use a state-package
 *                     entry point. Real evaluation lives in network.Policy.
 */
package space.u2re.cwsp.protocol.state;

import space.u2re.cwsp.protocol.packet.Packet;
import space.u2re.cwsp.protocol.packet.Types;

/**
 * State-package facade over {@link space.u2re.cwsp.protocol.network.Policy}.
 * Delegates to the canonical evaluator so all platforms share one impl.
 *
 * NOTE: The class is named {@code Policy} to match its file. The delegate is
 *       referenced fully-qualified to avoid a same-simple-name clash.
 */
public final class Policy {

    private Policy() {
    }

    public static space.u2re.cwsp.protocol.network.Policy.Decision evaluate(
            Packet packet, space.u2re.cwsp.protocol.network.Policy.Context context) {
        return space.u2re.cwsp.protocol.network.Policy.evaluate(packet, context);
    }

    public static space.u2re.cwsp.protocol.network.Policy.Decision evaluate(
            Packet packet, space.u2re.cwsp.protocol.network.Policy.Context context,
            space.u2re.cwsp.protocol.network.Policy.Overrides overrides) {
        return space.u2re.cwsp.protocol.network.Policy.evaluate(packet, context, overrides);
    }

    public static Types.PolicyClass classify(Packet packet) {
        return space.u2re.cwsp.protocol.network.Policy.classify(packet);
    }
}
